src/Dmr.Api/Controllers/DMRController.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Dmr.Api.Models;
+using Dmr.Api.Services.AsyncProcessor;
+using Microsoft.AspNetCore.Mvc;
+using System.Text;
+
+namespace Dmr.Api.Controllers
+{
+    [Route("/messages")]
+    [ApiController]
+    public class DmrController : ControllerBase
+    {
+        private readonly IAsyncProcessorService<Message> processor;
+
+        public DmrController(IAsyncProcessorService<Message> processor)
+        {
+            this.processor = processor;
+        }
+
+        /// <summary>
+        /// Accepts 'Post'ed messages with Buerokratt headers and an encoded/encrypted message body. 
+        /// </summary>
+        /// <param name="headers">Model bound headers</param>
+        /// <returns>204 Accepted or 400 BadRequest</returns>
+        [HttpPost]
+        public async Task<IActionResult> PostAsync([FromHeader] HeadersInput headers)
+        {
+            if (headers == null)
```

Comment:
could we add this in as well since the HeadersInput class in HeadersInput.cs seems to contain validation logic? 
```suggestion
            if (headers == null || !modelState.IsValid)
```

Reply:
Model validation occurs before this method is called so we shouldn't need to do this explicitly.  That null check was just belt and braces honestly - we shouldn't even be in here if there aren't any headers specified. I'll take that null check out - it shouldn't happen.